torch/nn/modules/module.py

```diff
@@ -461,3 +461,11 @@ def __repr__(self):
             tmpstr = tmpstr + '  (' + key + '): ' + modstr + '\n'
         tmpstr = tmpstr + ')'
         return tmpstr
+
+    def __dir__(self):
+        module_attrs = dir(self.__class__)
+        attrs = self.__dict__.keys()
+        parameters = self._parameters.keys()
+        modules = self._modules.keys()
+        buffers = self._buffers.keys()
+        return module_attrs + attrs + parameters + modules + buffers
```

Comment:
Can you please also wrap the result in `sorted`?  `__dir__` should always return sorted sequences